cmake/extensions.cmake

```diff
@@ -1472,6 +1472,13 @@ function(target_sources_ifdef feature_toggle target scope item)
   endif()
 endfunction()
 
+function(target_sources_if_dt_alias alias target scope item)
```

Comment:
Could you please add some description of this function.

I'm curios on one thing.
This adds a `_if_dt_alias` which internally uses `dt_prop(... "/aliases" ...)` but where will this end then ?
Will there be a dedicated function for each path / property someone might ask for ?

Like:
`target_sources_if_dt_status_ok()` calling `dt_prop(... PATH <path> STATUS ...)

or other dt_prop dependent calls with other combinations

I think I understand the benefit, but what i'm trying to understand is how many places you expect this to be used, or if  this function should be a bit more generic so we won't need to add `n` dt functions for each combination of `dt_prop()`.

Maybe:
`target_sources_if_dt_prop(<path> <prop> <target> <scope> <item>)`
which in your case could be invoked as:
`target_sources_if_dt_prop("/aliases" <prop> <target> <scope> <item>)` (`<prop>` being your current `<alias>`)

Reply:
I have reworked this PR substantially to address your concerns. Please revisit.